cli/bench_command.go

```diff
@@ -411,232 +665,1267 @@ func (c *benchCmd) bench(_ *fisk.ParseContext) error {
+		}()
+	}
+
+	if c.ack {
+		log.Printf("Defined durable explicitly acked push consumer: %s\n", c.consumerName)
+	} else {
+		log.Printf("Defined durable unacked push consumer: %s\n", c.consumerName)
+	}
+
+	subCounts := bench.MsgsPerClient(c.numMsg, c.numClients)
+
+	for i := 0; i < c.numClients; i++ {
+		nc, err := nats.Connect(opts().Config.ServerURL(), natsOpts()...)
+		if err != nil {
+			return fmt.Errorf("nats connection %d failed: %s", i, err)
+		}
+		defer nc.Close()
+
+		startwg.Add(1)
+		donewg.Add(1)
+
+		go c.runOldJSSubscriber(bm, nc, startwg, donewg, subCounts[i], BenchTypeOldJSPush)
+	}
+	startwg.Wait()
+
+	if !c.noProgress {
+		uiprogress.Start()
+	}
+
+	donewg.Wait()
+
+	bm.Close()
+
+	err = c.printResults(bm)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (c *benchCmd) oldjsPullAction(_ *fisk.ParseContext) error {
+	err := c.processActionArgs()
+	if err != nil {
+		return err
+	}
+
+	banner := c.generateBanner(BenchTypeOldJSPull)
+	log.Println(banner)
+	bm := bench.NewBenchmark("NATS", c.numClients, 0)
+
+	startwg := &sync.WaitGroup{}
+	donewg := &sync.WaitGroup{}
+
+	nc, err := nats.Connect(opts().Config.ServerURL(), natsOpts()...)
+	if err != nil {
+		log.Fatalf("NATS connection failed: %v", err)
+	}
+
+	js, err := nc.JetStream(append(jsOpts(), nats.MaxWait(c.jsTimeout))...)
+	if err != nil {
+		log.Fatalf("Couldn't get the JetStream context: %v", err)
+	}
+
+	ack := nats.AckNonePolicy
+	if c.ack {
+		ack = nats.AckExplicitPolicy
+	}
+
+	if c.consumerName == DefaultDurableConsumerName {
+		_, err = js.AddConsumer(c.streamName, &nats.ConsumerConfig{
+			Durable:       c.consumerName,
+			DeliverPolicy: nats.DeliverAllPolicy,
+			AckPolicy:     ack,
+			ReplayPolicy:  nats.ReplayInstantPolicy,
+			MaxAckPending: func(a int) int {
+				if a >= 10000 {
+					return a
+				} else {
+					return 10000
+				}
+			}(c.numClients * c.batchSize),
+		})
+		if err != nil {
+			log.Fatalf("Error creating the pull consumer: %v", err)
+		}
+		defer func() {
+			err := js.DeleteConsumer(c.streamName, c.consumerName)
+			if err != nil {
+				log.Printf("Error deleting the pull consumer on stream %s: %v", c.streamName, err)
+			}
+			log.Printf("Deleted durable consumer: %s\n", c.consumerName)
+		}()
+		log.Printf("Defined durable pull consumer: %s\n", c.consumerName)
+	}
+
+	subCounts := bench.MsgsPerClient(c.numMsg, c.numClients)
+
+	for i := 0; i < c.numClients; i++ {
+		nc, err := nats.Connect(opts().Config.ServerURL(), natsOpts()...)
+		if err != nil {
+			return fmt.Errorf("nats connection %d failed: %s", i, err)
+		}
+		defer nc.Close()
+
+		startwg.Add(1)
+		donewg.Add(1)
+
+		go c.runOldJSSubscriber(bm, nc, startwg, donewg, subCounts[i], BenchTypeOldJSPull)
+	}
+	startwg.Wait()
+
+	if !c.noProgress {
+		uiprogress.Start()
+	}
+
+	donewg.Wait()
+
+	bm.Close()
+
+	err = c.printResults(bm)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func getSubscribeSubject(c *benchCmd) string {
```

Comment:
please attach all these to the `banchCmd` so we dont polute the global function space so much with undocumented and unclearly named functions

Reply:
Sure, done.